### DISCUSSION TITLE
R1, R2 and R3 values

### DISCUSSION THREAD

**Discussion by AlexandreUSA** [Dilbert66/esphome-vistaECP#42] [Q&A] [answered]
Just saw that you changed the R2 value to be 27K from 33K. What drove the change? Did you get feedback on field issues? If you have a 12V system, the 27K and 10K voltage divider provides the 3.24V level needed for the D1 or D5. However, are we not closer to 14V level on the vista systems? I think that last time I measure the voltage (red to black), it was at 13.5V. Would the yellow/green bus at the same level? Using 27K or lower may put you above the 3.6V max. Maybe 30K is the middle ground.

For others, if they don't have the correct resistor values, we are looking for a voltage divider with a ratio of ~ 27% at 12V  or R3/(R2+R3).

Also, you mentioned R1 at 220 Ohm as the min value. Any specific reason for this value? Looking to keep the current at ~10ma for efficiency for the 4N26?  I am asking the question as I had 200 Ohm available. I did add a 21 Ohm in series but would be easier with only one...

**Comment by Dilbert66** [comment]
I was getting weak signals on the monitor line for some modules which caused incorrect values. The new value gave me consistent traffic. That might be related to the fact that I'm using a lower voltage input transformer. You do have a point though that for a normal system the higher kohm values should be correct. I will revert it on the schematic with a caveat for users to change it if they need to.  As to R!, we need to keep the drive current from the ESP pins under their rated max which if I remember correctly is around 12ma.    You should be fine with a 200ohm.   The current is calculated as 3.3 - 0.7 (diode approx voltage drop) / 200ohm = 13ma.   All those values are approximate of course.

**Reply by AlexandreUSA** [reply]
The fun of the project. For the 4N26, is the forward voltage 1.2V at 10mA? is it not then 3.3V-1.2V so with a 200ohm, you would be at 10ma and with the 220ohm, just under 10?

By the way, if you are updating the readme file, you may want to mention that the Vista system (20p, 21ip, etc) only provide 600mA for all keypads, motion sensors, etc. I decided to drop using the 12V line as my power supply for this project due to this. I already have 3 keypads (max 140mA), 1 extension board (max 100mA), plus the motion sensors, I am probably at the max level already but motions are not all on at the same time. You do mention using a 800mA power supply if going USB, that seems a lot. Do you know what the ESP32 uses for this project?

**Reply by Dilbert66** [reply]
That's a very good point about the power useage.   A separate supply would definitively be recommended for the esp board.   The issue with the esp power useage depends on the wifi useage of course.   Since it is constantly sending data and not sitting idle, it could use a bit.  I'd say a max maybe of 300ma but I have not measured it.  If I said, 800ma, that's overkill. I should revise that. I think I went by a use case I saw somewhere that was similar. 

For the 4n26 forward voltage, you are correct about the forward voltage but I tend to use .7 to give a bit of a safety window for real world differences. But as you noted, 1.2 gives you a very good value at 200ohm so you'll be fine.

**Reply by AlexandreUSA** [reply]
800mA seems to be peak value based on this website: https://lastminuteengineers.com/esp32-sleep-modes-power-consumption/
> ESP32 Active Mode
> The normal mode is also known as Active Mode. In this mode all the features of the chip are active.
> 
> As the active mode keeps everything (especially the WiFi module, the Processing Cores and the Bluetooth module) ON at all times, the chip requires more than 240mA current to operate. Also we observed that if you use both WiFi and Bluetooth functions together, sometimes high power spikes appear (biggest was 790mA).

```
Mode | Power Consumption
Wi-Fi Tx packet 13dBm~21dBm | 160~260mA
Wi-Fi/BT Tx packet 0dBm | 120mA
Wi-Fi/BT Rx and listening | 80~90mA
```

However, we can probably use the 240mA as the baseline. A standard 500mA is probably enough. I do use a 1A one so all good here.

**Comment by Dilbert66** [comment, marked as answer]
FYI, to clarify, I double checked the signal voltage on the yellow/green line is a regulated 12 volt and not the red line voltage.  So you can have 14volt on the red line, but it will still be 12 volt on the signal lines.  The divider values are correct for 12v.  You can check it yourself with an oscilloscope.

**Reply by AlexandreUSA** [reply]
Thanks for checking. I do not have a scope but your explanation for the signal lines makes sense. I may be overthinking it but do you see any issue with using my 13.6V (Red line) to drive the Green line (using the 4N26)? What about the TX monitoring voltage divider? I am sure how the green line works with the monitoring.

**Reply by Dilbert66** [reply]
On my system, with a red line voltage of 13.7 volts and using a 220ohm resistor for the 4n25, I get pulses of about 11.5volt amplitude. With a 180ohm resistor, the pulses get to around 13.5 volt.   I didnt test with a 200 ohm but would guess it would push it to about 12.5 volts  which is within the safe range of the tx line voltage divider.  So if you don't go lower than 200ohm for the optocoupler driver resistor, you will be fine.
